# IS 362 – Week 5 Assignment
## Your task in this week’s assignment is to answer three questions:
1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

Use the information in the .CSV files found at https://github.com/hadley/nycflights13/tree/master/data-raw for your source data Write your code using Python and pandas, and deliver it in a Jupyter notebook. You should post your notebook in a GitHub repository, and send me a link to the notebook.


First we import the `airports.csv` file into a Dataframe

In [1]:
import pandas as pd

In [2]:
airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
airports 

faa                                  name        lat         lon   alt  \
0     04G                     Lansdowne Airport  41.130472  -80.619583  1044   
1     06A         Moton Field Municipal Airport  32.460572  -85.680028   264   
2     06C                   Schaumburg Regional  41.989341  -88.101243   801   
3     06N                       Randall Airport  41.431912  -74.391561   523   
4     09J                 Jekyll Island Airport  31.074472  -81.427778    11   
5     0A9        Elizabethton Municipal Airport  36.371222  -82.173417  1593   
6     0G6               Williams County Airport  41.467306  -84.506778   730   
7     0G7         Finger Lakes Regional Airport  42.883565  -76.781232   492   
8     0P2          Shoestring Aviation Airfield  39.794824  -76.647191  1000   
9     0S9                 Jefferson County Intl  48.053809 -122.810644   108   
10    0W3                Harford County Airport  39.566838  -76.202403   409   
11    10C                    Galt Field Airport  42.402889  -88.375111   875   
12    17G  Port Bucyrus-Crawford County Airport  40.781556  -82.974806  1003   
13    19A                Jackson County Airport  34.175864  -83.561597   951   
14    1A3         Martin Campbell Field Airport  35.015806  -84.346833  1789   
15    1B9                   Mansfield Municipal  42.000133  -71.196771   122   
16    1C9                  Frazier Lake Airpark  54.013333 -124.768333   152   
17    1CS            Clow International Airport  41.695974  -88.129231   670   
18    1G3                    Kent State Airport  41.151389  -81.415111  1134   
19    1G4             Grand Canyon West Airport  35.899904 -113.815674  4813   
20    1H2            Effingham Memorial Airport  39.070000  -88.534000   585   
21    1OH                       Fortman Airport  40.555325  -84.386619   885   
22    1RL                 Point Roberts Airpark  48.979722 -123.078889    10   
23    23M                             Clarke CO  32.051700  -88.443400   320   
24    24C                   Lowell City Airport  42.953920  -85.343906   681   
25    24J               Suwannee County Airport  30.300125  -83.024694   104   
26    25D                   Forest Lake Airport  45.247746  -92.994385   925   
27    29D                    Grove City Airport  41.146028  -80.167750  1371   
28    2A0                    Mark Anton Airport  35.486250  -84.931083   718   
29    2B2                   Plum Island Airport  42.795361  -70.839444    11   
...   ...                                   ...        ...         ...   ...   
1428  X59                    Valkaria Municipal  27.960861  -80.558333    26   
1429  XFL                Flagler County Airport  29.282100  -81.121200    33   
1430  XNA                  NW Arkansas Regional  36.281869  -94.306811  1287   
1431  XZK            Amherst Amtrak Station AMM  42.375000  -72.511389   258   
1432  Y51                     Municipal Airport  43.579360  -90.896474  1292   
1433  Y72                          Bloyer Field  43.976222  -90.480611   966   
1434  YAK                               Yakutat  59.301200 -139.393700    33   
1435  YIP                            Willow Run  42.237928  -83.530408   716   
1436  YKM  Yakima Air Terminal McAllister Field  46.568200 -120.544000  1095   
1437  YKN                           Chan Gurney  42.871100  -97.396900  1200   
1438  YNG                Youngstown Warren Rgnl  41.260736  -80.679097  1196   
1439  YUM                   Yuma Mcas Yuma Intl  32.656578 -114.605980   216   
1440  Z84                                 Clear  64.301203 -149.120144   552   
1441  ZBP                          Penn Station  39.307222  -76.615556    66   
1442  ZFV          Philadelphia 30th St Station  39.955700  -75.182000     0   
1443  ZPH                     Municipal Airport  28.228056  -82.155833    90   
1444  ZRA           Atlantic City Rail Terminal  39.366500  -74.442000     8   
1445  ZRD                         Train Station  37.534300  -77.429450    26   
1446  

now that we have our Dataframe we should serch it for NaN values

In [3]:
airports.isnull().sum()

faa      0
name     0
lat      0
lon      0
alt      0
tz       0
dst      0
tzone    3
dtype: int64

Luckily we only have 3 tzones  values that are`NaN` 

*** 
Lets pull up those 3 rows to decide our aproch better.

In [4]:
airports[airports.tzone.isnull()]

faa                                  name        lat         lon  alt  \
417   EEN               Dillant Hopkins Airport  72.270833   42.898333  149   
815   LRO  Mount Pleasant Regional-Faison Field  32.538700  -79.469700   12   
1434  YAK                               Yakutat  59.301200 -139.393700   33   

      tz dst tzone  
417   -5   A   NaN  
815   -5   A   NaN  
1434  -9   A   NaN

here are the 3 entries that contain `NaN` values

***
In order to make changed with out changing our original data, we will make a copy of the `airports` Dataframe. We will call it `airportscopy`

 Now we will change the `NaN` values in our `airportscopy`, and change those `NaN` values to `'UNKNOWN'`

In [5]:
airportscopy = airports.copy()
airportscopy['tzone'].fillna(value='UNKNOWN', inplace=True)
airportscopy

faa                                  name        lat         lon   alt  \
0     04G                     Lansdowne Airport  41.130472  -80.619583  1044   
1     06A         Moton Field Municipal Airport  32.460572  -85.680028   264   
2     06C                   Schaumburg Regional  41.989341  -88.101243   801   
3     06N                       Randall Airport  41.431912  -74.391561   523   
4     09J                 Jekyll Island Airport  31.074472  -81.427778    11   
5     0A9        Elizabethton Municipal Airport  36.371222  -82.173417  1593   
6     0G6               Williams County Airport  41.467306  -84.506778   730   
7     0G7         Finger Lakes Regional Airport  42.883565  -76.781232   492   
8     0P2          Shoestring Aviation Airfield  39.794824  -76.647191  1000   
9     0S9                 Jefferson County Intl  48.053809 -122.810644   108   
10    0W3                Harford County Airport  39.566838  -76.202403   409   
11    10C                    Galt Field Airport  42.402889  -88.375111   875   
12    17G  Port Bucyrus-Crawford County Airport  40.781556  -82.974806  1003   
13    19A                Jackson County Airport  34.175864  -83.561597   951   
14    1A3         Martin Campbell Field Airport  35.015806  -84.346833  1789   
15    1B9                   Mansfield Municipal  42.000133  -71.196771   122   
16    1C9                  Frazier Lake Airpark  54.013333 -124.768333   152   
17    1CS            Clow International Airport  41.695974  -88.129231   670   
18    1G3                    Kent State Airport  41.151389  -81.415111  1134   
19    1G4             Grand Canyon West Airport  35.899904 -113.815674  4813   
20    1H2            Effingham Memorial Airport  39.070000  -88.534000   585   
21    1OH                       Fortman Airport  40.555325  -84.386619   885   
22    1RL                 Point Roberts Airpark  48.979722 -123.078889    10   
23    23M                             Clarke CO  32.051700  -88.443400   320   
24    24C                   Lowell City Airport  42.953920  -85.343906   681   
25    24J               Suwannee County Airport  30.300125  -83.024694   104   
26    25D                   Forest Lake Airport  45.247746  -92.994385   925   
27    29D                    Grove City Airport  41.146028  -80.167750  1371   
28    2A0                    Mark Anton Airport  35.486250  -84.931083   718   
29    2B2                   Plum Island Airport  42.795361  -70.839444    11   
...   ...                                   ...        ...         ...   ...   
1428  X59                    Valkaria Municipal  27.960861  -80.558333    26   
1429  XFL                Flagler County Airport  29.282100  -81.121200    33   
1430  XNA                  NW Arkansas Regional  36.281869  -94.306811  1287   
1431  XZK            Amherst Amtrak Station AMM  42.375000  -72.511389   258   
1432  Y51                     Municipal Airport  43.579360  -90.896474  1292   
1433  Y72                          Bloyer Field  43.976222  -90.480611   966   
1434  YAK                               Yakutat  59.301200 -139.393700    33   
1435  YIP                            Willow Run  42.237928  -83.530408   716   
1436  YKM  Yakima Air Terminal McAllister Field  46.568200 -120.544000  1095   
1437  YKN                           Chan Gurney  42.871100  -97.396900  1200   
1438  YNG                Youngstown Warren Rgnl  41.260736  -80.679097  1196   
1439  YUM                   Yuma Mcas Yuma Intl  32.656578 -114.605980   216   
1440  Z84                                 Clear  64.301203 -149.120144   552   
1441  ZBP                          Penn Station  39.307222  -76.615556    66   
1442  ZFV          Philadelphia 30th St Station  39.955700  -75.182000     0   
1443  ZPH                     Municipal Airport  28.228056  -82.155833    90   
1444  ZRA           Atlantic City Rail Terminal  39.366500  -74.442000     8   
1445  ZRD                         Train Station  37.534300  -77.429450    26   
1446  

now since all the values in tzones are strings we can serch tzones  for all airports tzones that are in 'America'.
***

we will create a new dataframe `us_airports` to represent just results that have a tzone that contains the word `'America'` in it.

In [6]:
us_airports = airportscopy[airportscopy['tzone'].str.contains("America")]
us_airports

faa                                  name        lat         lon   alt  \
0     04G                     Lansdowne Airport  41.130472  -80.619583  1044   
1     06A         Moton Field Municipal Airport  32.460572  -85.680028   264   
2     06C                   Schaumburg Regional  41.989341  -88.101243   801   
3     06N                       Randall Airport  41.431912  -74.391561   523   
4     09J                 Jekyll Island Airport  31.074472  -81.427778    11   
5     0A9        Elizabethton Municipal Airport  36.371222  -82.173417  1593   
6     0G6               Williams County Airport  41.467306  -84.506778   730   
7     0G7         Finger Lakes Regional Airport  42.883565  -76.781232   492   
8     0P2          Shoestring Aviation Airfield  39.794824  -76.647191  1000   
9     0S9                 Jefferson County Intl  48.053809 -122.810644   108   
10    0W3                Harford County Airport  39.566838  -76.202403   409   
11    10C                    Galt Field Airport  42.402889  -88.375111   875   
12    17G  Port Bucyrus-Crawford County Airport  40.781556  -82.974806  1003   
13    19A                Jackson County Airport  34.175864  -83.561597   951   
14    1A3         Martin Campbell Field Airport  35.015806  -84.346833  1789   
15    1B9                   Mansfield Municipal  42.000133  -71.196771   122   
16    1C9                  Frazier Lake Airpark  54.013333 -124.768333   152   
17    1CS            Clow International Airport  41.695974  -88.129231   670   
18    1G3                    Kent State Airport  41.151389  -81.415111  1134   
19    1G4             Grand Canyon West Airport  35.899904 -113.815674  4813   
20    1H2            Effingham Memorial Airport  39.070000  -88.534000   585   
21    1OH                       Fortman Airport  40.555325  -84.386619   885   
22    1RL                 Point Roberts Airpark  48.979722 -123.078889    10   
23    23M                             Clarke CO  32.051700  -88.443400   320   
24    24C                   Lowell City Airport  42.953920  -85.343906   681   
25    24J               Suwannee County Airport  30.300125  -83.024694   104   
26    25D                   Forest Lake Airport  45.247746  -92.994385   925   
27    29D                    Grove City Airport  41.146028  -80.167750  1371   
28    2A0                    Mark Anton Airport  35.486250  -84.931083   718   
29    2B2                   Plum Island Airport  42.795361  -70.839444    11   
...   ...                                   ...        ...         ...   ...   
1427  X49                South Lakeland Airport  27.933358  -82.043974   110   
1428  X59                    Valkaria Municipal  27.960861  -80.558333    26   
1429  XFL                Flagler County Airport  29.282100  -81.121200    33   
1430  XNA                  NW Arkansas Regional  36.281869  -94.306811  1287   
1431  XZK            Amherst Amtrak Station AMM  42.375000  -72.511389   258   
1432  Y51                     Municipal Airport  43.579360  -90.896474  1292   
1433  Y72                          Bloyer Field  43.976222  -90.480611   966   
1435  YIP                            Willow Run  42.237928  -83.530408   716   
1436  YKM  Yakima Air Terminal McAllister Field  46.568200 -120.544000  1095   
1437  YKN                           Chan Gurney  42.871100  -97.396900  1200   
1438  YNG                Youngstown Warren Rgnl  41.260736  -80.679097  1196   
1439  YUM                   Yuma Mcas Yuma Intl  32.656578 -114.605980   216   
1440  Z84                                 Clear  64.301203 -149.120144   552   
1441  ZBP                          Penn Station  39.307222  -76.615556    66   
1442  ZFV          Philadelphia 30th St Station  39.955700  -75.182000     0   
1443  ZPH                     Municipal Airport  28.228056  -82.155833    90   
1444  ZRA           Atlantic City Rail Terminal  39.366500  -74.442000     8   
1445  ZRD                         Train Station  37.534300  -77.429450    26   
1446  

Lets compare our results and see how many rows were removed

In [7]:
rows_removed = len(airports) - len(us_airports)
rows_removed

23

Since we lost alot of rows we should check our value counts of `tzone` in our `airportscopy` Dataframe to se exsactly what `tzones` were removed when we created `us_airports`.

In [8]:
airportscopy['tzone'].value_counts(dropna=False)

America/New_York       519
America/Chicago        342
America/Anchorage      239
America/Los_Angeles    176
America/Denver         119
America/Phoenix         38
Pacific/Honolulu        18
UNKNOWN                  3
America/Vancouver        2
Asia/Chongqing           2
Name: tzone, dtype: int64

For the sake of this assiment, we will only keep the data we can confirm is in America based on the data we are given. 

We can see we have tzone Asia/Chongqing that we know is not in the us so tthose can be removed

we also see we have Pacific/Honolulu now this dose not say america even though hawaii is in the pacific and is in the us , we will still remove it becuse not all airports in the pacific are in america and we are only looking up northern most and eastern most airports so we can remove the Pacific/Honolulu results as they are west of the US

Also we have the tzone we created that were NaN and are now 'UNKNOWN'

so if we compare this to list to what is in `us_airports`

In [9]:
us_airports['tzone'].value_counts(dropna=False)

America/New_York       519
America/Chicago        342
America/Anchorage      239
America/Los_Angeles    176
America/Denver         119
America/Phoenix         38
America/Vancouver        2
Name: tzone, dtype: int64

I am comfortable using this to answere our questions 1 & 2

For question **1. What is the northernmost airport in the United States?** in order to find our northern most airport we will look for the top 5 rsults with the largest `'lat'` results in `us_airports` we will cal this new result `northern_most_airports`

In [10]:
northern_most_airports = us_airports.sort_values(by = 'lat', ascending = False).head()
northern_most_airports

faa                                        name        lat         lon  \
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   

      alt  tz dst              tzone  
230    44  -9   A  America/Anchorage  
110    41  -9   A  America/Anchorage  
708    35  -9   A  America/Anchorage  
152    96  -9   A  America/Anchorage  
1363   67  -9   A  America/Anchorage

According to a quick search or our top result [Google Map Search](https://www.google.com/maps/place/Wiley+Post-Will+Rogers+Memorial+Airport/@63.083625,-132.8824806,4z/data=!4m5!3m4!1s0x0:0x936de8c8fd2d094b!8m2!3d71.2874212!4d-156.7802701) search or our top result `Wiley Post Will Rogers Mem` we can see that it is at the very top of Alaska the northern most state in America so this seems to be true.
from IPython.core.display import display, HTML
![pic](https://photos.wikimapia.org/p/00/01/29/78/51_big.jpg)


## So, the Answer to question *1. What is the northernmost airport in the United States?* is:
## **Wiley Post Will Rogers Mem**
***

For question **2. What is the easternmost airport in the United States?** we just need to do a little modification to our `northern_most_airports` search. we will swap out `lat` for `lon`, meaning longitued, and rename our new result `eastern_most_airports` and this should give us our answere.

In [11]:
eastern_most_airports = us_airports.sort_values(by = 'lon', ascending = False).head()
eastern_most_airports

faa                                 name        lat         lon  alt  \
1290  SYA                         Eareckson As  52.712275  174.113620   98   
444   EPM           Eastport Municipal Airport  44.910111  -67.012694   45   
624   HUL                         Houlton Intl  46.123083  -67.792056  489   
259   CAR                         Caribou Muni  46.871500  -68.017917  626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958  -68.044797  534   

      tz dst              tzone  
1290  -9   A  America/Anchorage  
444   -5   A   America/New_York  
624   -5   A   America/New_York  
259   -5   A   America/New_York  
1101  -5   A   America/New_York

Our top result is `Eareckson As`. The problem I see with this is according to the `tzone` in the dataframe, this airport is in the Alaska time zone, and alaska is located west of the United states. A quick internet serch and  [Wikipedia](https://en.wikipedia.org/wiki/Eareckson_Air_Station) confirms this.
So I must assume the `lon` in this entry for `Eareckson As` is incorect. We will move on to the next row result which is `Eastport Municipal Airport`.


Once we do a quick [Google Maps Serch](https://www.google.com/maps/place/Eastport+Municipal+Airport-EPM/@40.1860431,-101.0731964,4.96z/data=!4m5!3m4!1s0x0:0xe8e7c4805bb5f1e8!8m2!3d44.9099796!4d-67.0119367) we see that `Eastport Municipal Airport` is in Main which is so far east in the US that it is near the border of Canada.
![Eastport Runway pic](https://www.funplacestofly.com/photos/airportlowe85242.jpg)

## So, the Answer to question *2. What is the easternmost airport in the United States?* is:
## **Eastport Municipal Airport**
***

For question **3. On February 12th, 2013, which New York area airport had the windiest weather?** we will switch csv files and use the `weather.csv` file in the git hub repository. we will import this csv file into a dataframe named `weather`.

In [12]:
weather = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')
weather

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0        EWR  2013      1    1     1  39.02  26.06  59.37     270.0   
1        EWR  2013      1    1     2  39.02  26.96  61.63     250.0   
2        EWR  2013      1    1     3  39.02  28.04  64.43     240.0   
3        EWR  2013      1    1     4  39.92  28.04  62.21     250.0   
4        EWR  2013      1    1     5  39.02  28.04  64.43     260.0   
5        EWR  2013      1    1     6  37.94  28.04  67.21     240.0   
6        EWR  2013      1    1     7  39.02  28.04  64.43     240.0   
7        EWR  2013      1    1     8  39.92  28.04  62.21     250.0   
8        EWR  2013      1    1     9  39.92  28.04  62.21     260.0   
9        EWR  2013      1    1    10  41.00  28.04  59.65     260.0   
10       EWR  2013      1    1    11  41.00  26.96  57.06     260.0   
11       EWR  2013      1    1    13  39.20  28.40  69.67     330.0   
12       EWR  2013      1    1    14  39.02  24.08  54.68     280.0   
13       EWR  2013      1    1    15  37.94  24.08  57.04     290.0   
14       EWR  2013      1    1    16  37.04  19.94  49.62     300.0   
15       EWR  2013      1    1    17  35.96  19.04  49.83     330.0   
16       EWR  2013      1    1    18  33.98  15.08  45.43     310.0   
17       EWR  2013      1    1    19  33.08  12.92  42.84     320.0   
18       EWR  2013      1    1    20  32.00  15.08  49.19     310.0   
19       EWR  2013      1    1    21  30.02  12.92  48.48     320.0   
20       EWR  2013      1    1    22  28.94  12.02  48.69     320.0   
21       EWR  2013      1    1    23  28.04  10.94  48.15     310.0   
22       EWR  2013      1    2     0  26.96  10.94  50.34     310.0   
23       EWR  2013      1    2     1  26.06  10.94  52.25     330.0   
24       EWR  2013      1    2     2  24.98  10.94  54.65     330.0   
25       EWR  2013      1    2     3  24.08   8.96  51.93     320.0   
26       EWR  2013      1    2     4  24.08   8.96  51.93     330.0   
27       EWR  2013      1    2     5  24.08   8.96  51.93     330.0   
28       EWR  2013      1    2     6  24.08   8.96  51.93     310.0   
29       EWR  2013      1    2     7  24.98  10.04  52.50     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26085    LGA  2013     12   29    13  42.80  37.94  88.76      70.0   
26086    LGA  2013     12   29    14  41.00  37.94  93.19      60.0   
26087    LGA  2013     12   29    15  41.00  39.02  92.59      40.0   
26088    LGA  2013     12   29    16  41.00  37.94  88.76     350.0   
26089    LGA  2013     12   29    17  44.06  41.00  93.24     350.0   
26090    LGA  2013     12   29    18  42.08  39.02  88.81     330.0   
26091    LGA  2013     12   29    19  42.80  37.94  85.13     320.0   
26092    LGA  2013     12   29    20  42.08  37.94  86.89     320.0   
26093    LGA  2013     12   29    21  42.80  37.40  82.17     320.0   
26094    LGA  2013     12   29    22  42.98  37.04  79.38     300.0   
26095    LGA  2013     12   29    23  42.98  35.06  73.39     310.0   
26096    LGA  2013     12   30     0  42.08  33.98  72.78     320.0   
26097    LGA  2013     12   30     1  42.08  33.98  72.78     250.0   
26098    LGA  2013     12   30     2  41.00  33.98  75.88     240.0   
26099    LGA  2013     12   30     3  42.98  33.98  70.30     270.0   
26100    LGA  2013     12   30     4  41.00  33.08  73.19       0.0   
26101    LGA  2013     12   30     5  42.98  33.08  67.81     250.0   
26102    LGA  2013     12   30     6  42.98  33.98  70.30     230.0   
26103    LGA  2013     12   30     7  44.06  35.06  70.42     240.0   
26104    LGA  2013     12   30     8  44.06  33.98  67.45     260.0   
26105    LGA  2013     12   30     9  44.06  33.08  65.07     260.0   
26106    LGA  2013     12   30    10  42.98  33.80  70.28     330.0   
26107    LGA  2013     12   30    11  41.00  28.40  62.21     340.0   
26108    LGA  2013     12   30    12  37.94  23.00  54.51     330.0   
26109    LGA  2013     12   30

***
Now that we have all the weather in a dataframe,  we can filter it to only return us the results from february 12th, 2013 we will name this `feb_12_2013_weather`.


In [13]:
feb_12_2013_weather = weather[(weather.year == 2013) & (weather.day == 12) & (weather.month == 2)]
feb_12_2013_weather

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007     EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010     EWR  2013      2   12     4  39.02  26.96  64.29     280.0   
1011     EWR  2013      2   12     5  39.92  26.96  60.29     270.0   
1012     EWR  2013      2   12     6  39.02  26.96  61.63     270.0   
1013     EWR  2013      2   12     7  39.92  24.98  54.81     260.0   
1014     EWR  2013      2   12     8  42.08  26.60  60.29     250.0   
1015     EWR  2013      2   12     9  41.90  24.98  50.77     290.0   
1016     EWR  2013      2   12    10  42.08  26.06  52.73     260.0   
1017     EWR  2013      2   12    11  42.98  24.98  48.70     260.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
1019     EWR  2013      2   12    13  46.04  26.06  45.33     280.0   
1020     EWR  2013      2   12    14  44.96  26.06  47.23     280.0   
1021     EWR  2013      2   12    15  42.98  24.98  48.70     300.0   
1022     EWR  2013      2   12    16  42.98  24.08  46.91     270.0   
1023     EWR  2013      2   12    17  41.00  24.08  50.63     260.0   
1024     EWR  2013      2   12    18  39.92  24.08  52.79     270.0   
1025     EWR  2013      2   12    19  39.92  24.08  52.79     270.0   
1026     EWR  2013      2   12    20  37.94  24.08  57.04     280.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
1028     EWR  2013      2   12    22  35.96  23.00  58.94     260.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
9710     JFK  2013      2   12     1  39.02  35.96  88.66     250.0   
9711     JFK  2013      2   12     2  41.00  28.94  61.89     280.0   
9712     JFK  2013      2   12     3  41.00  26.06  54.97     290.0   
9713     JFK  2013      2   12     4  39.92  24.98  54.81     280.0   
9714     JFK  2013      2   12     5  39.02  26.96  61.63     260.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
9727     JFK  2013      2   12    18  41.00  24.08  50.63     270.0   
9728     JFK  2013      2   12    19  39.92  23.00  50.46     280.0   
9729     JFK  2013      2   12    20  39.02  21.92  49.93     290.0   
9730     JFK  2013      2   12    21  37.94  21.92  52.09     300.0   
9731     JFK  2013      2   12    22  37.94  21.92  52.09     290.0   
9732     JFK  2013      2   12    23  37.04  21.92  53.97     280.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
18416    LGA  2013      2   12     1  44.06  30.92  59.64     270.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
18418    LGA  2013      2   12     3  42.08  26.06  52.73     290.0   
18419    LGA  2013      2   12     4  41.00  24.98  52.56     270.0   
18420    LGA  2013      2   12     5  41.00  24.98  52.56     290.0   
18421    LGA  2013      2   12     6  42.08  24.08  48.56     270.0   
18422    LGA  2013      2   12     7  41.00  24.08  50.63     260.0   
18423    LGA  2013      2   12     8  42.08  23.00  46.41     260.0   
18424    LGA  2013      2   12     9  42.08  23.00  46.41     260.0   
18425    LGA  2013      2   12    10  42.98  23.00  44.83     270.0   
18426    LGA  2013      2   12    11  42.98  23.00  44.83     270.0   
18427    LGA  2013      2   12    12  44.06  23.00  43.02     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
18430    LGA  2013      2   12    15  44.06  21.92  41.10     260.0   
18431    LGA  2013      2   12    16  42.98  24.08  46.91     300.0   
18432    LGA  2013      2   12    17  42.98  23.00  44.83     260.0   
18433    LGA  2013      2   12

Now that we have all the weather from airports for feb. 12th 2013 we will take the top 5 with the windiest weather, in a result called `windiest_weather`

In [14]:

windiest_weather = feb_12_2013_weather.sort_values(by='wind_speed', ascending = False).head()
windiest_weather

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z

Now that we have the top 5 `windiest_weather` for `feb_12_2013_weather` we can see that EWR which stands for Newark Airport, had the top wind speed for feb 12th 2013, with a wind speed of 1048. However, While EWR is a New York area airport, it is not in New York. Additionally, if 1048 mph wind speed was to be belived, it would have blown EWR off the map. So, this leads me to belive this data can not be correect, so i will go with the next result.

## So for question 3. On February 12th, 2013, which New York area airport had the windiest weather?
 I will say **LGA, Laguardia Airport, had the windiest weather*** with a recorded wind speed of 23 mph.